### PR TITLE
Name validation backend

### PR DESCRIPTION
**Pull request by kidCaulfield**
Response to issue #76 

I have added validation to the API for child names and sponsor names.

Using a regular expression to exclude most immediate special characters on a north American keyboard.

Special characters accepted are the following:
'-' for hyphenated names
' ' for spaces between names
'.' for sponsor names with titles such as 'Dr.' ect.

Names are now required, empty strings will not be excepted.

If validation fails, the API with return an error object that includes a useful error message for frontend developers to display to the user.

### CODE REVIEW

**Comment by kidCaulfield** [comment]
I looked over the Error and this is failing because the tests are failing the validation I set up in the API. Which shows the name validation is working.

**Comment by kidCaulfield** [comment]
Finished updating tests for the backend name validation for wishes.

File: api/src/db/Wish.js [hunk 1]
```diff
@@ -1,19 +1,34 @@
...
       hometown: String,
       illness: String,
       age: Number
     },
     type: { type: String, enum: ['go', 'meet', 'be', 'have'] },
     details: String,
     sponsor: {
-      name: String,
+      name: { type: String,
+        validate: [
```
Comment (crstamps2) [inline api/src/db/Wish.js]: I think that sponsors can definitely have number's and/or symbols in their name. For instance, Make-a-Wish itself has hyphens in it's name. Thoughts?

File: api/src/routes/wish.js [hunk 2]
```diff
@@ -127,7 +129,7 @@ wishRouter
       req.body,
       { new: true },
       (err, wish) => {
-        if (err) return res.status(500).send(err)
+        if (err) return res.status(409).send(err)
```
Comment (crstamps2) [inline api/src/routes/wish.js]: Hi @kidCaulfield, thank you for the contribution. Can you add a test to capture this error case. I realize we might not have had one for a 500 error, but we appreciate helping fill these gaps.

File: api/src/app.test.js [hunk 3]
```diff
@@ -82,4 +95,74 @@ describe('a wish', () => {
...
+          logo: 'K',
+          links: []
+        }
+      })
+
+    expect(response.status).toBe(422)
+    expect(response.body.errors['sponsor.name'].message).toBe('Sponsor\'s name required')
+  })
+
+  it('should return 422 and error message for nameValidation \'Name must contain only letters of the alphabet\'', async () => {
```
Comment (crstamps2) [inline api/src/app.test.js]: can you also through in a test to check again symbols too (for kid's name only, not sponsor)?

File: api/src/db/Wish.js (hunk 1 above)
Comment (kidCaulfield) [reply api/src/db/Wish.js]: Hyphens are in the allowed special characters however numbers are currently not allow. I see what your saying though. If a company is a sponsor they may have numbers in their name. I could add numbers to the sponsor allowed special characters or just pull the regex validation completely for symbols and number in the case of "sponsors".

File: api/src/routes/wish.js (hunk 2 above)
Comment (kidCaulfield) [reply api/src/routes/wish.js]: I can put the 500 check back in.

File: api/src/app.test.js (hunk 3 above)
Comment (kidCaulfield) [reply api/src/app.test.js]: Not sure what you mean. I have a written a test to check for symbols in kids name. I can remove the test for symbols in sponsors names if that what you mean?

**Comment by kidCaulfield** [comment]
I've updated my pull as requested. Covering the issues requested and addressing the delete error case test.

File: api/src/app.test.js (hunk 3 above)
Comment (WRMilling) [reply api/src/app.test.js]: Hey @kidCaulfield, I believe he means throw in a test that ensure you can use diacritics in a child's name like `Matías` or `Nicolás`. If it works with those extra characters (just adding some to the happy path test should be sufficient) I can finalize the review and merge.

**Comment by kidCaulfield** [comment]
@WRMilling and @crstamps2 added the test for diacritics in the child's. I think that covers all the requested changes.

**Comment by WRMilling** [comment]
@kidCaulfield Thank you again for your contribution! Looks good in my local testing and all CI checks have passed, merging in shortly.

**Pull request merged**